### ISSUE TITLE
Add an option of `disable_localhost`

### ISSUE THREAD

**Issue by eyalzek** [bblimke/webmock#695]
Currently `allow_localhost` exists, but the opposite of it doesn't. My use case is when using `chefspec` and stubbing requests going to localhost. The problem is that `berkshelf` is trying to resolve cookbook dependencies, making requests to `https://supermarket.chef.io/universe` and a few more other domains.

I tried `WebMock.enable!(except: [:berkshelf])` which didn't work, I also tried to disable webmock for chef supermarket but the problem there is that it's making further connection to S3 and several other sites, so it's not really feasible to cover all of them.

I ended up doing:
`WebMock.disable_net_connect!(allow: %r{^(?!.*localhost).*$})`

to disable webmock for everything but localhost. This looks really bad. I might be missing a better way achieve this, otherwise I think it's a feature that would be nice to have.

**Comment by bblimke** [comment]
@eyalzek it's rather a rare scenario that someone want's to disable only localhost, that's why there is no support for that.

Why don't you list the domains you want to allow connections to as an array passed to allow: ?

**Comment by eyalzek** [comment]
Because if I add the domain that it's complaining about it'll complain about another and then another and so on. Also, I can't be sure that the sources they use will remain the same it the future, so it makes more sense for me to disable everything but `localhost`...

**Comment by bblimke** [comment]
How about: `WebMock.disable_net_connect!(allow: lambda{|uri| uri.host.name != 'localhost' })`

**Comment by eyalzek** [comment]
This gives:

```
Failure/Error: WebMock.disable_net_connect!(allow: lambda{|uri| uri.host.name != 'localhost' })

NoMethodError:
  undefined method `name' for "supermarket.chef.io":String
```

But changing `uri.host.name` -> `uri.host` seems to work :+1:

**Comment by kingdonb** [comment]
I was just looking for an alternative to `allow_localhost: true` because some of my devs are using vagrant on Windows with Chrome, and their Chrome browsers are on the other side of 10.0.2.2:4444 -- wow!  How about adding some documentation about the extra power-versions of this feature?

I forked and made a local copy of your gem to solve my issue, I just added 10.0.2.2 to the definition of localhost and went on using `allow_localhost: true`.  But from this issue report discussion, it's clear that there are more powerful ways to reduce the impact of disable_net_connect!

If you need me to submit a separate issue, I will do it?

If you're strapped for time, I could possibly take a look at this in more depth and try to generate some more comprehensive documentation for you.  It might be better if you did it, but let me know.

Also, looks like this issue was resolved for the submitter and can be closed.

**Comment by bblimke** [comment]
@kingdonb I'd be more than happy to accept a pull request with the improved documentation :)

**Comment by kingdonb** [comment]
@bblimke Well I've just gone and had another look at the README, turns out it's already quite extensive on this point!

I think I wanted disable_net_connect!(allow_localhost: true, allow: '10.0.2.2')

**Comment by bblimke** [comment]
@kingdonb great!

The documentation is not that great though, if we had to discuss the solution here :)

In general WebMock deserves much better documentation than the current one page README.

**Comment by kingdonb** [comment]
I agree with that.  I am kind of a nascent tech-blogger, I'd love to contribute something.  Whether that's updated documentation in the form of a Wiki, or some reorganization of the README into separate documents with a TOC... or a blog post about how to use WebMock!

Will put a pin in this, my app is going to production next weekend and there is a lot to do between now and then.
